utils/azure/identity/default_credentials.cc

```diff
@@ -133,8 +133,11 @@ future<default_credentials::credentials_opt> default_credentials::get_credential
     auto creds = std::make_unique<managed_identity_credentials>(_imds_endpoint, _logctx);
     try {
         co_await creds->get_access_token(resource_uri);
-    } catch (timed_out_error&) {
-        az_creds_logger.info("[{}] Failed to connect to IMDS.", *this);
+    } catch (timed_out_error& e) {
+        az_creds_logger.info("[{}] Failed to connect to IMDS: timed out", *this);
+        co_return std::nullopt;
+    } catch (std::system_error& e) {
+        az_creds_logger.info("[{}] System error while accessing IMDS: {}", *this, e.what());
```

Comment:
Should this not verify _which_ system error you got? Technically, it could be something that this logic would/should classify as propagation-level.

Reply:
Invalid address stuff, anything related to general system error? In general, when handling system error we do (and should) check the error codes as well, and only special treat the based on this...
As usual, none of these other cases will ever happen, but...